java/arcs/sdk/Entity.kt

```diff
@@ -11,11 +11,11 @@
 
 package arcs.sdk
 
-abstract class Entity {
-    var internalId = ""
-    abstract fun schemaHash(): String
-    abstract fun isSet(): Boolean
-    abstract fun getFieldsNotSet(): List<String>
+interface Entity {
+    var internalId: String
```

Comment:
If you wanted this to still have the default value of `""`, you can provide a default getter in the interface.

```kotlin
var internalId: String
    get() = ""
```

Reply:
I tried that, but it gave me this error:

```
error: property in an interface cannot have a backing field
    var internalId: String
    ^
```

I think it has to be a val, but then you can't reassign it.

Anyway, I'm pretty sure it's only needed in the wasm code, so I can move it into WasmEntity in a follow up PR.